testing/cli/test_sct_deepseg.py

```diff
@@ -38,24 +37,67 @@ def test_model_dict():
         assert 'default' in value
 
 
-@pytest.mark.parametrize('fname_image, fname_seg_manual, fname_out, task', [
+@pytest.mark.parametrize('fname_image, fname_seg_manual, fname_out, task, thr', [
     (sct_test_path('t2s', 't2s.nii.gz'),
      sct_test_path('t2s', 't2s_seg-deepseg.nii.gz'),
      't2s_seg_deepseg.nii.gz',
-     'seg_sc_t2star'),
+     'seg_sc_t2star',
+     0.9),
+    (sct_test_path('t2', 't2.nii.gz'),
+     sct_test_path('t2', 't2_seg-manual.nii.gz'),
+     't2_seg_deepseg.nii.gz',
+     'seg_sc_contrast_agnostic',
+     0.5),
+    (sct_test_path('t2', 't2.nii.gz'),
+     None,  # TODO: implement integrity test (for now, just checking if output segmentation file exists)
+     't2_seg_deepseg.nii.gz',
+     'seg_spinal_rootlets_t2w',
+     0.5),
 ])
-def test_segment_nifti(fname_image, fname_seg_manual, fname_out, task,
+def test_segment_nifti(fname_image, fname_seg_manual, fname_out, task, thr,
                        tmp_path):
     """
     Uses the locally-installed sct_testing_data
     """
     # Ignore warnings from ivadomed model source code changing
     warnings.filterwarnings("ignore", category=SourceChangeWarning)
     fname_out = str(tmp_path/fname_out)  # tmp_path for automatic cleanup
-    sct_deepseg.main(['-i', fname_image, '-task', task, '-o', fname_out, '-thr', str(0.9)])
-    # TODO: implement integrity test (for now, just checking if output segmentation file exists)
+    sct_deepseg.main(['-i', fname_image, '-task', task, '-o', fname_out, '-thr', str(thr)])
     # Make sure output file exists
     assert os.path.isfile(fname_out)
-    # Compare with ground-truth segmentation
-    assert np.all(nibabel.load(fname_out).get_fdata() ==
-                  nibabel.load(fname_seg_manual).get_fdata()[..., 0])
+    # Compare with ground-truth segmentation if provided
+    if fname_seg_manual:
+        im_seg = Image(fname_out)
+        im_seg_manual = Image(fname_seg_manual)
+        im_seg_manual.data = im_seg_manual.data.squeeze()  # FIXME: t2s_seg-deepseg.nii.gz has an extra axis
+        dice_segmentation = compute_dice(im_seg, im_seg_manual, mode='3d', zboundaries=False)
```

Comment:
When updating the existing test, I found that the ground truth file for the existing `T2s` test had an extra axis, and needed a `squeeze` in order to run the `compute_dict` function without error.

I think we should be able to fix this on the `sct_testing_data` level? But, maybe could be done in separate issue/PR.

Reply:
Resolved upstream!